### PR TITLE
Uninstall openshift-logging for ocp 4.2

### PR DESCRIPTION
**Pull request by Shrivaibavi**
Signed-off-by: Shrivaibavi Raghaventhiran <sraghave@redhat.com>

### CODE REVIEW

**Comment by Shrivaibavi** [comment]
```Collecting Cluster versions
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.0, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/sraghave/cl_deploy/ocs-ci, inifile: pytest.ini
plugins: logger-0.5.0, reportportal-1.0.5, metadata-1.8.0, html-1.21.1, marker-bugzilla-0.9.1.dev2
collected 1 item                                                                                                                                                                                                  

tests/test_deployment.py::TestLogging::test_create_new_project_to_verify_logging 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
17:47:14 - MainThread - tests.conftest - INFO - All logs located at /tmp/ocs-ci-logs-1565353005
17:47:14 - MainThread - /home/sraghave/cl_deploy/ocs-ci/ocs_ci/deployment/factory.py - INFO - Deployment key = aws_ipi
17:47:14 - MainThread - /home/sraghave/cl_deploy/ocs-ci/ocs_ci/deployment/factory.py - INFO - Current deployment platform: AWS,deployment type: ipi
17:47:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ./bin/openshift-install version
17:47:14 - MainThread - ocs_ci.utility.utils - INFO - OpenShift Installer version: ./bin/openshift-install v4.2.0-201907300219-dirty
                                                      built from commit 9323138c52b773e31b7c98d8a934f9ddaa6ccaee
                                                      release image registry.svc.ci.openshift.org/ocp/release@sha256:eb93f85c7caec2d7603f384f1a0ae6c004fc3dcdff90c957b074c4ff1b790e2b
17:47:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ./bin/oc version
17:47:16 - MainThread - ocs_ci.utility.utils - INFO - OpenShift Client version: Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-alpha.0-34-g0d02ccfc", GitCommit:"0d02ccfccbfbabe8be161db4dbb1f42bf00cf7c2", GitTreeState:"clean", BuildDate:"2019-07-28T20:08:21Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.0+edda291", GitCommit:"edda291", GitTreeState:"clean", BuildDate:"2019-08-05T23:58:35Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      OpenShift Version: 4.2.0-0.nightly-2019-08-06-195545
17:47:16 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/sraghave/Downloads/auth/kubeconfig
17:47:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
17:47:18 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Access to cluster is OK!
17:47:18 - MainThread - /home/sraghave/cl_deploy/ocs-ci/ocs_ci/deployment/aws.py - WARNING - OCP cluster is already running, skipping installation
17:47:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get CephCluster  -o yaml
17:47:20 - MainThread - /home/sraghave/cl_deploy/ocs-ci/ocs_ci/deployment/deployment.py - WARNING - OCS cluster already exists
PASSED                                                                                                                                                                                                      [100%]
------------------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------------------
17:47:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace cluster-logging-operator-6ccbf596c-dm6t4
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-4x23c1br-1-696b6b7c5-9f8db
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-4x23c1br-2-f6c79b9cb-4fvmr
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-4x23c1br-3-5949c6557c-k795v
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-4bcjg
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-n4d4v
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-n68jj
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-rldcc
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-vdw48
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-wvfm8
17:47:23 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace kibana-5bd5544f87-gkhgr
17:47:23 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig delete ClusterLogging instance
17:47:26 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:35 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod kibana-5bd5544f87-gkhgr does not exist in the given namespace
17:47:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:39 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-4bcjg exists in the given namespace
17:47:39 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:42 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-n4d4v exists in the given namespace
17:47:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:44 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-n68jj exists in the given namespace
17:47:44 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:47 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-rldcc exists in the given namespace
17:47:47 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:50 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-vdw48 exists in the given namespace
17:47:50 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:53 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-wvfm8 exists in the given namespace
17:47:53 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Pod  -o yaml
17:47:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig get Namespaces  -o yaml
17:47:59 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The namespaces exists, Deleting.....
17:47:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/Downloads/auth/kubeconfig delete Namespaces openshift-logging
17:48:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-operators-redhat --kubeconfig /home/sraghave/Downloads/auth/kubeconfig delete Namespaces openshift-operators-redhat
17:48:49 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The projects got deleted successfully


============================================================================================ 1 passed in 94.78 seconds ============================================================================================```

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -613,3 +613,26 @@ def get_pod_obj(name, namespace=None):
     ocp_dict = ocp_obj.get(resource_name=name)
     pod_obj = Pod(**ocp_dict)
     return pod_obj
+
+
+def verify_pod_exists(namespace=None, pod_name=None):
+    """
+    Verifies if given pod in present in the given namespace
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/pod.py]: *is* present in...

File: ocs_ci/utility/uninstall_openshift_logging.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        logger.info(f"Pods running in the openshift-logging namespace {pod.name}")
+
+    # Deleting the clusterlogging instance
+    clusterlogging_obj = ocp.OCP(
+        kind='ClusterLogging', namespace='openshift-logging'
+    )
+    assert clusterlogging_obj.delete(resource_name='instance')
+
+    # Verifying the pods if exists after deleting instance
+    pod_list = [
```
Comment (RazTamir) [inline ocs_ci/utility/uninstall_openshift_logging.py]: pod_list is still the same list as before the deletion.
you'll need to call get_all_pods() again to get a fresh list

**Comment by vasukulkarni** [comment]
@Shrivaibavi couple more comments to fix and needs rebase, thanks!

File: ocs_ci/utility/uninstall_openshift_logging.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    # Deleting the projects
+    openshift_logging_namespace = ocp.OCP(
+        kind=constants.NAMESPACES, namespace='openshift-logging'
+    )
+    openshift_operators_redhat_namespace = ocp.OCP(
+        kind=constants.NAMESPACES, namespace='openshift-operators-redhat'
+    )
+
+    if openshift_logging_namespace.get() or openshift_operators_redhat_namespace.get():
```
Comment (keesturam) [inline ocs_ci/utility/uninstall_openshift_logging.py]: If one exists and the other doesn't, this is going to fail. so pls do this one by one. 

```
if openshift_logging_namespace.get():
    assert openshift_logging_namespace.delete(resource_name='openshift-logging')
if openshift_operators_redhat_namespace.get():
    assert openshift_operators_redhat_namespace.delete(resource_name='openshift-operators-redhat')
```

File: ocs_ci/utility/uninstall_openshift_logging.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+def uninstall_cluster_logging():
+    """
+    Function to uninstall cluster-logging from the cluster
+    1. Deletes the project "openshift-logging"
+        & "openshift-operators-redhat"
+    2. Deletes the CSC resource "cluster-logging-operator"
```
Comment (keesturam) [inline ocs_ci/utility/uninstall_openshift_logging.py]: '2' is achieved by deleting the openshift_operators_redhat project I believe. can you update the doc string accordingly?

File: ocs_ci/ocs/resources/pod.py (hunk 1 above)
Comment (Shrivaibavi) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/ocs/resources/pod.py [hunk 5]
```diff
@@ -613,3 +613,26 @@ def get_pod_obj(name, namespace=None):
...
+
+    Returns:
+        bool: True if pod exists in the namespace, False
+        otherwise
+
+    """
+    pod_list = get_all_pods(namespace=namespace)
+    for pod in pod_list:
+        if pod.name == pod_name:
+            logger.info(f"Pod {pod.name} exists in the given namespace")
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/utility/uninstall_openshift_logging.py (hunk 3 above)
Comment (Shrivaibavi) [reply ocs_ci/utility/uninstall_openshift_logging.py]: Done

File: ocs_ci/utility/uninstall_openshift_logging.py (hunk 4 above)
Comment (Shrivaibavi) [reply ocs_ci/utility/uninstall_openshift_logging.py]: Done

File: ocs_ci/ocs/resources/pod.py [hunk 6]
```diff
@@ -613,3 +613,26 @@ def get_pod_obj(name, namespace=None):
     ocp_dict = ocp_obj.get(resource_name=name)
     pod_obj = Pod(**ocp_dict)
     return pod_obj
+
+
+def verify_pod_exists(namespace=None, pod_name=None):
```
[the remark this answers is not in the document]
Comment (Shrivaibavi) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/utility/uninstall_openshift_logging.py (hunk 2 above)
Comment (Shrivaibavi) [reply ocs_ci/utility/uninstall_openshift_logging.py]: I checked it. Initially the pod list contained 11 items including cluster-logging-operator, after the removal of one item from the list it became 10 excluding cluster-logging-operator. Attaching the rough logs for the reference
```17:50:26 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - 11
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace cluster-logging-operator-dd48849c-dn4rw
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-1-55f577bb95-5x2zc
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-2-65b4d4d4fb-42njl
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-3-58fd8b7546-zl9zn
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-5nw8c
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-drz85
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-gptkt
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-jnfms
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-xg2b6
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-z6c5x
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace kibana-5bd5544f87-vrssw
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - 10
17:50:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:32 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-1-55f577bb95-5x2zc exists in the given namespace
17:50:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:35 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-2-65b4d4d4fb-42njl exists in the given namespace
17:50:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:38 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-3-58fd8b7546-zl9zn exists in the given namespace
17:50:38 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:41 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-5nw8c exists in the given namespace
17:50:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:43 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-drz85 exists in the given namespace
17:50:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:46 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-gptkt exists in the given namespace
17:50:46 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:49 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-jnfms exists in the given namespace
17:50:49 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:52 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-xg2b6 exists in the given namespace
17:50:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:54 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-z6c5x exists in the given namespace
17:50:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:57 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod kibana-5bd5544f87-vrssw exists in the given namespace
17:50:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Namespaces  -o yaml
17:50:59 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The projects got deleted successfully```

**Comment by keesturam** [comment]
@Shrivaibavi I've removed the verified label. pls retest with the new changes you've made today.

**Comment by Shrivaibavi** [comment]
```Collecting Cluster versions
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.0, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/sraghave/new_vai/ocs-ci, inifile: pytest.ini
plugins: logger-0.5.0, reportportal-1.0.5, metadata-1.8.0, html-1.21.1, marker-bugzilla-0.9.1.dev2
collected 1 item                                                                                                                                                                                                  

tests/test_uninstall.py::test_fun 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
19:17:02 - MainThread - tests.conftest - INFO - All logs located at /tmp/ocs-ci-logs-1566913602
19:17:02 - MainThread - ocs_ci.deployment.factory - INFO - Deployment key = aws_ipi
19:17:02 - MainThread - ocs_ci.deployment.factory - INFO - Current deployment platform: AWS,deployment type: ipi
19:17:03 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ./bin/oc version
19:17:05 - MainThread - ocs_ci.utility.utils - INFO - OpenShift Client version: Client Version: version.Info{Major:"", Minor:"", GitVersion:"v4.2.0-alpha.0-2-g8fdb79e5", GitCommit:"8fdb79e549651c0f3c91d54349715309b5d149d3", GitTreeState:"clean", BuildDate:"2019-08-05T22:00:15Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.0+f95ec38", GitCommit:"f95ec38", GitTreeState:"clean", BuildDate:"2019-08-22T23:10:14Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      OpenShift Version: 4.2.0-0.nightly-2019-08-23-034826
19:17:05 - MainThread - ocs_ci.utility.utils - INFO - The rook directory /home/sraghave/new_vai/ocs-ci/data/rook already exists, ocs-ci will skip the clone of rook repository.
19:17:05 - MainThread - ocs_ci.utility.utils - INFO - Fetching latest changes from rook repository.
19:17:05 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git fetch --all
19:17:07 - MainThread - ocs_ci.utility.utils - INFO - Checkout rook repository to specific branch: master
19:17:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git checkout master
19:17:07 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Already on 'master'
19:17:07 - MainThread - ocs_ci.utility.utils - INFO - Reset branch: master with latet changes
19:17:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git reset --hard origin/master
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
19:17:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace cluster-logging-operator-5f6fc55fd6-xrvpc
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-aahlhw8a-1-67c4b75bf4-z8r7k
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-aahlhw8a-2-84b48d89-9lkh7
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-aahlhw8a-3-557c8f6d6b-298g7
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-82vfg
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-8z4rm
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-g7wrp
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-jt68k
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-t8g24
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-zwtkz
19:17:09 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace kibana-5bd5544f87-nbtxm
19:17:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete ClusterLogging instance
19:17:11 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
19:17:14 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Namespaces  -o yaml
19:17:16 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete Namespaces openshift-logging
19:17:58 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The namespace openshift-logging got deleted successfully
19:17:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-operators-redhat --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Namespaces  -o yaml
19:18:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-operators-redhat --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete Namespaces openshift-operators-redhat
19:18:24 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The project openshift-opertors-redhat got deleted successfully
PASSED                                                                                                                                                                                                      [100%]

============================================================================================ 1 passed in 82.62 seconds ============================================================================================```

File: ocs_ci/utility/uninstall_openshift_logging.py (hunk 2 above)
Comment (RazTamir) [reply ocs_ci/utility/uninstall_openshift_logging.py]: from a pythonic POV that not possible.
I'm not sure how that's possible that pod_list was changed bug please call get_all_pods again to refresh the list

**Comment by Shrivaibavi** [comment]
```Collecting Cluster versions
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.0, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/sraghave/new_vai/ocs-ci, inifile: pytest.ini
plugins: logger-0.5.0, reportportal-1.0.5, metadata-1.8.0, html-1.21.1, marker-bugzilla-0.9.1.dev2
collected 1 item                                                                                                                                                                                                  

tests/test_uninstall.py::test_fun 
------------------------------------------------------------------------------------------------- live log setup --------------------------------------------------------------------------------------------------
13:09:53 - MainThread - tests.conftest - INFO - All logs located at /tmp/ocs-ci-logs-1566977971
13:09:53 - MainThread - ocs_ci.deployment.factory - INFO - Deployment key = aws_ipi
13:09:53 - MainThread - ocs_ci.deployment.factory - INFO - Current deployment platform: AWS,deployment type: ipi
13:09:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ./bin/oc version
13:09:56 - MainThread - ocs_ci.utility.utils - INFO - OpenShift Client version: Client Version: version.Info{Major:"", Minor:"", GitVersion:"v4.2.0-alpha.0-2-g8fdb79e5", GitCommit:"8fdb79e549651c0f3c91d54349715309b5d149d3", GitTreeState:"clean", BuildDate:"2019-08-05T22:00:15Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.0+f95ec38", GitCommit:"f95ec38", GitTreeState:"clean", BuildDate:"2019-08-22T23:10:14Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"linux/amd64"}
                                                      OpenShift Version: 4.2.0-0.nightly-2019-08-23-034826
13:09:56 - MainThread - ocs_ci.utility.utils - INFO - The rook directory /home/sraghave/new_vai/ocs-ci/data/rook already exists, ocs-ci will skip the clone of rook repository.
13:09:56 - MainThread - ocs_ci.utility.utils - INFO - Fetching latest changes from rook repository.
13:09:56 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git fetch --all
13:09:58 - MainThread - ocs_ci.utility.utils - INFO - Checkout rook repository to specific branch: master
13:09:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git checkout master
13:09:58 - MainThread - ocs_ci.utility.utils - WARNING - Command warning: Already on 'master'
13:09:58 - MainThread - ocs_ci.utility.utils - INFO - Reset branch: master with latet changes
13:09:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: git reset --hard origin/master
-------------------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------------------
13:09:58 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace cluster-logging-operator-dbb7fd996-mblw7
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-t9gtu8zw-1-7c97fff98-kfth4
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-t9gtu8zw-2-68f6c49fd7-654gd
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-t9gtu8zw-3-765dffc9fc-v5kq9
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-4ht26
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-6fvjh
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-7l2xz
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-ch7p5
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-q5kvm
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-ss9nq
13:10:02 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace kibana-5bd5544f87-lmr9t
13:10:02 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete ClusterLogging instance
13:10:03 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
13:10:06 - MainThread - ocs_ci.utility.retry - WARNING - , Retrying in 30 seconds...
13:10:36 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
13:10:39 - MainThread - ocs_ci.utility.retry - WARNING - , Retrying in 60 seconds...
13:11:39 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Pod  -o yaml
13:11:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Namespaces  -o yaml
13:11:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete Namespaces openshift-logging
13:12:07 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The namespace openshift-logging got deleted successfully
13:12:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-operators-redhat --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig get Namespaces  -o yaml
13:12:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-operators-redhat --kubeconfig /home/sraghave/test-ocs/auth/kubeconfig delete Namespaces openshift-operators-redhat
13:12:33 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The project openshift-opertors-redhat got deleted successfully
PASSED                                                                                                                                                                                                      [100%]

=========================================================================================== 1 passed in 160.34 seconds ============================================================================================```

File: ocs_ci/utility/uninstall_openshift_logging.py (hunk 2 above)
Comment (Shrivaibavi) [reply ocs_ci/utility/uninstall_openshift_logging.py]: @RazTamir Done.

**Comment by vasukulkarni** [comment]
Lets add the test too and get your PR's merged!

**Pull request merged**